text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+* Signing crates.io index - this is covered by other proposals, but is discussed in the context of TUF later in this document.
+* Extending or altering the Identity approach of Sigstore, i.e. OIDC based
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Crate signatures and verifications are a way to tie the authenticity and integrity of a crate based on public key cryptography. The implication is that, as an end user, you can be sure that the crates that you download from crates.io have not been tampered with, and that they are signed by the correct owner of the package.
+
+The process of `signing` involves generating a signature for a crate, using a private key, and attaching that signature when publishing a crate. The process of `verification` involves downloading a crate (using cargo) and using a _trusted_ public key to verify the authenticity (the signer) and integrity (the contents) of a crate.
+
+To simplify key management for users, cargo supports using [Sigstore](https://sigstore.dev) as a way to generate ephemeral keys based on an identity, and as an immutable log that can be monitored and audited in the event of a compromise. By default, a publicly available instance ensures that you do not need to manage this infrastructure yourself. If you are running your own crate registry, Sigstore instance and identify provider, you can configure cargo to use that.
+
+Verification is accomplished through an API interaction with the relevant Rekor service. This occurs at artifact submission time to the relevant crates registry and is also how the consumer will verify the artifact whether manually or in a run-time scenario.
+
+## Implementation sequencing
+
+## Phase 1
+
+* Establish the required metadata structures and modifications to crates.io to accommodate the Sigstore signing process.
+* Modify cargo to add the Sigstore specific capabilities for signing and verification, and configuration options to require signed artifacts from crates.io during dependency pulls, etc.
+* Modify crates.io to accept the additional metadata attributes from the signing operation.
+* Provide inspection / verification capabilities in the crates.io UI. for Sigstore signed crates
+
+## Phase 2
+
+* TUF (The Update Framework) - Phase 2 TUF AuthZ Architecture
+   * The TUF protocol addresses a set of defined attacks and threat models specific to software distribution systems. It has the capability of providing a strong AuthZ implementation for any OSS project with respect to control of the ability to generate an artifact. For example, it is used by the Sigstore project itself to instantiate and manage the Sigstore Public Root. See: [public Sigstore instance](https://blog.sigstore.dev/a-new-kind-of-trust-root-f11eeeed92ef). This article does a good job of showing the options.
+* Make Sigstore signed crates mandatory.
+* Start marking legacy / unmaintained unsigned crates as unsafe
```

Comment:
I don't know that it is a correct use of `unsafe`. What's the proof obligation for safety in this case? Is there any? It feels like it's being abuse for things that are dispreferred.

Not to mention, it sounds like it would be a breaking change, so I don't think this is viable.

Reply:
I also dislike the term unsafe here. To me, unsigned would be a better choice.